### PR TITLE
adding to lodash best practices

### PR DESCRIPTION
**Pull request by lukehansell**
Adding examples to the lodash best practice based on discussions in the front-end guild.
- [x] +1
- [ ] +2

### CODE REVIEW

File: general-javascript-best-practices.md [hunk 1]
```diff
@@ -149,6 +149,27 @@ Use instead of underscore, due to correct semantic versioning, better performanc
...
+```
+
+**Bad:**
+```javascript
+const _ = require('lodash');
+```
+
+As an alternative (and where Lodash is not currently a dependency) you could install and require each Lodash package individually:
+
+**Better:**
```
Comment (Joezo) [inline general-javascript-best-practices.md]: I disagree with this being the "better" option. I think this is only better if you're using a version of lodash that doesn't let you `require('lodash/func')`. In v4 for example you can just do `require('lodash/includes')`. There is no point adding extra multiple dependencies when we can just install it all in one go. 

I think we should always use the _good_ option. It's probably worth adding that when using the _good_ option you can't use `_.chain`. Or at least i'm yet to find a way of using it in that way.

Comment (lukehansell) [inline general-javascript-best-practices.md]: based on this and other feedback I'll remove the `better` option

**Comment by hx-markterry** [comment]
ok with me :bank:

File: general-javascript-best-practices.md [hunk 2]
```diff
@@ -149,6 +149,18 @@ Use instead of underscore, due to correct semantic versioning, better performanc
 
 Please only require the methods you need rather than the whole library in order to keep build sizes at a minimum.
 
+**Good:**
```
Comment (Joezo) [inline general-javascript-best-practices.md:152]: I think we should prefer that people don't use `_.chain` in clientside code. There is no way of using that with this approach. We also lose all the benefits of doing it this way if someone requires the whole of lodash just to use `_.chain`. Thoughts?

Comment (lukehansell) [inline general-javascript-best-practices.md:152]: I agree if it's not possible to use `_.chain` in this way but all other methods can still be required. I'll merge this and then we can add another PR to discuss and advise on `_.chain`

**Comment by lukehansell** [comment]
any further objections to me merging this?

**Comment by Joezo** [comment]
not from me :)

**Pull request merged**